docs/content/preview/architecture/key-concepts.md

```diff
@@ -0,0 +1,135 @@
+
+CDC is a software design pattern used in database systems to capture and propagate data changes from one database to another in real-time or near real-time. YugabyteDB supports transactional CDC guaranteeing changes across tables are captured together. This enables use cases like real-time analytics, data warehousing, operational data replication, and event-driven architectures. {{<link "../docdb-replication/change-data-capture/">}}
+
+## Cluster
+
+A cluster is a group of [nodes](#node) on which YugabyteDB is deployed. The table data is distributed across the various [nodes](#node) in the cluster. Typically used as [*Primary cluster*](#primary-cluster) and [*Read replica cluster*](#read-replica-cluster).
+
+{{<note>}}
+Sometimes the term *cluster* is used interchangeably with the term *universe*. However, the two are not always equivalent, as described in [Universe](#universe).
+{{</note>}}
+
+## DocDB
+
+DocDB is the underlying document storage engine of YugabyteDB and is built on top of a highly customized and optimized verison of [RocksDB](http://rocksdb.org/) {{<link "../docdb">}}
+
+## Fault domain
+
+A fault domain is a potential point of failure. Examples of fault domains would be nodes, racks, zones, or entire regions. {{<link "../../../explore/fault-tolerance/#fault-domains">}}
+## Master server
+
+The [YB-Master](../yb-master/) service is responsible for keeping system metadata, coordinating system-wide operations, such as creating, altering, and dropping tables, as well as initiating maintenance operations such as load balancing. {{<link "yb-master">}}
+
+{{<tip>}}
+The master server is also typically referred as just **master**
+{{</tip>}}
+
+## MVCC
+
+MVCC stands for Multiversion Concurrency Control. It is a concurrency control method used by YugabyteDB to provide access to data in a way that allows concurrent queries and updates without causing conflicts. {{<link "../../transactions/transactions-overview/#mvcc-using-hybrid-time">}}
+
+## Namespace
+
+A namespace refers to a logical grouping or container for related database objects, such as tables, views, indexes, and other database constructs. Namespaces help organize and separate these objects, preventing naming conflicts and providing a way to control access and permissions.
+
+A namespace in YSQL is referred to as a database and is logically identical to a namespace in other RDBMS (such as PostgreSQL) and a namespace in YCQL is referred to as a keyspace and is logically identical to a keyspace in Apache Cassandra's CQL.
+
+## Node
+
+A node is a virtual machine, physical machine, or container on which YugabyteDB is deployed.
+
+## Leader balancing
+
+YugabyteDB tries to keep the no.of leaders evenly distributed across the [nodes](#node) in a cluster to ensure an even distribution of load.
+
+## Primary cluster
+
+A primary cluster can perform both writes and reads. Replication between [nodes](#node) in a primary cluster is performed synchronously.
```

Comment:
As opposed to what? Why even make the distinction?

Reply:
As opposed to a read replica cluster. Will add that in